### PR TITLE
Improve Linux part, implement functions in libsddc and fix debug stacktrace read in firmware

### PR DESCRIPTION
**Pull request by kpd200081**
I would use libsddc for RX888 mk2 on Linux, but facing that a lot of functions not implemented and streaming was not working for me.

There are still some bugs and place for improvements, but I decide to share my progress with others.

### CODE REVIEW

**Comment by ik1xpv** [comment]
Hi Pavel, many thanks for sharing your code progress.   I think Linux support is very important, unfortunately my personal knowledge of Linux is low.  Fortunately Franco, Howard, Hayati have directed the code evolution to go in the direction of trying to offer support to the Linux world as well.
Oscar

File: Core/fft_mt_r2iq.h
Comment (howard0su) [inline Core/fft_mt_r2iq.h:10]: what is the issue when having multi threads?

File: libsddc/libsddc.cpp
Comment (howard0su) [inline libsddc/libsddc.cpp:205]: I believe RX888 also supports pga

File: libsddc/libsddc.cpp
Comment (howard0su) [inline libsddc/libsddc.cpp:291]: i am not sure why we use double as freq while uint64_t should better serve the purpose in the original sddc interface? @hayguen @fventuri

File: libsddc/libsddc.cpp
Comment (howard0su) [inline libsddc/libsddc.cpp:391]: fabsf for float

**Review by howard0su — CHANGES_REQUESTED** [review]

**Comment by howard0su** [comment]
appreciate your PR.

File: Core/fft_mt_r2iq.h
Comment (kpd200081) [reply Core/fft_mt_r2iq.h:10]: Segmentation fault happened for me somewhere in libfft if I enable multi threads (even in unittest). I will try to compile recent libfft from source and see what changed.

But anyway, r2iq not currently used in libsddc, I just forgot to revert the changes.

File: libsddc/libsddc.cpp
Comment (kpd200081) [reply libsddc/libsddc.cpp:205]: I copied this from here https://github.com/ik1xpv/ExtIO_sddc/blob/master/ExtIO_sddc/tdialog.cpp#L75

File: libsddc/libsddc.cpp
Comment (kpd200081) [reply libsddc/libsddc.cpp:391]: Oh, sure, thanks for noticing

**Comment by kpd200081** [comment]
Thanks for attention here, Howard and Oscar

**Comment by howard0su** [comment]
> Thanks for attention here, Howard and Oscar

My pleasure. please cleanup and we can commit. I will also build a package for the libsddc for Linux/Windows for application to consume.

File: Core/fft_mt_r2iq.h
Comment (kpd200081) [reply Core/fft_mt_r2iq.h]: I reverted this to original state

File: libsddc/libsddc.cpp
Comment (kpd200081) [reply libsddc/libsddc.cpp]: fixed

File: Core/arch/linux/FX3handler.cpp
Comment (kpd200081) [inline Core/arch/linux/FX3handler.cpp:37]: Here I was trying to avoid freezes when using device controls during active streaming. Now it freezes less, but not perfect, if anyone can suggest a better way to fix it I'll be glad.

Comment (howard0su) [reply Core/arch/linux/FX3handler.cpp:37]: I looked at libusb code once. I am not expert. It is my suspicion that we didn't handle multi thread well. I suggest this page:
http://libusb.sourceforge.net/api-1.0/libusb_mtasync.html

Comment (kpd200081) [reply Core/arch/linux/FX3handler.cpp:37]: Oh yes, it looks like we have a situation described at the end of the article on your link.  Thanks, I'll try to figure it out.

Comment (kpd200081) [reply Core/arch/linux/FX3handler.cpp:37]: I tried to figure out what happens in bulk transfers with libusb, but with no success. I found that streaming only works if there is a control transfers to the device (such as reading a debug trace), but under such conditions, all other controls are delayed by about a second. Without the control transfers bulk transfers timeouts, it looks like it tries to work when setting the transfer timeout to lowest possible 1ms, but this is not enough for full functionality anyway.
My competence is not enough to resolve this issue.

Comment (kpd200081) [reply Core/arch/linux/FX3handler.cpp]: Looks like I near solved this. Now it's possible to normally use streaming on linux, but sometimes streaming stops from libusb timeouts.
Also I was forced to reduce QUEUE_SIZE to 16 from 32, because with 32 I wasn't able get stable streaming with libusb.
Unfortunately, I can not test is something broke on Windows or not.

Comment (ik1xpv) [reply Core/arch/linux/FX3handler.cpp]: Pavel thanks for the code. It compiles in windows. In release at first look it seems to run ok while in debug it stops streaming after start up.
We have to understand what's happening  before merging it to master.
Howard in the while is making big change in #199 and #200.

Comment (kpd200081) [reply Core/arch/linux/FX3handler.cpp]: Hm, it's looking as in debug we have enabled reading debug from fx3 and this not works...
It's strange that without my change in firmware, reading not working good on Linux...
Anyway debug read on linux currently not working because of timeouts in streaming so I will just revert this change.

Comment (kpd200081) [reply Core/arch/linux/FX3handler.cpp]: I reverted changes in firmware, please check when you will have a time.

File: Core/config.cpp
Comment (howard0su) [inline Core/config.cpp:5]: can we revert the changes in this file?

File: Core/config.h
Comment (howard0su) [inline Core/config.h:4]: all good changes but not related to linux usb. can we do these changes in a separate change?

File: libsddc/libsddc.cpp
Comment (howard0su) [inline libsddc/libsddc.cpp:434]: sample rate is related to adc rate.

**Review by howard0su — APPROVED** [review]

File: libsddc/libsddc.cpp
Comment (kpd200081) [reply libsddc/libsddc.cpp:434]: Currently only two ADC rates supported in libsddc (64 and 128 MHz) and this relation handled here https://github.com/ik1xpv/ExtIO_sddc/blob/master/Core/RadioHandler.cpp#L154, isn't it?

File: Core/config.h
Comment (kpd200081) [reply Core/config.h]: Reverted all except QUEUE_SIZE change

**Comment by kpd200081** [comment]
Please do not merge this yet, I found problem in streaming stopping.

**Pull request closed without merging**
